howso/update_cases.amlg

```diff
@@ -218,17 +218,102 @@
 		))
 
 		(if case_ids
-			(map
-				(lambda
-					(call !UpdateCaseWithHistory (assoc
-						case_id (current_value 1)
-						features features
-						feature_values feature_values
-						unencoded_feature_values unencoded_feature_values
-						session session
-					))
+			(seq
+				(map
+					(lambda
+						(call !UpdateCaseWithHistory (assoc
+							case_id (current_value 1)
+							features features
+							feature_values feature_values
+							unencoded_feature_values unencoded_feature_values
+							session session
+						))
+					)
+					case_ids
+				)
+
+				;update derived features if necessary
+				(let
+					(assoc
+						affected_derived_features
+							;any features edited that are the source feature of a non-TS derived feature
+							(filter
+								(lambda
+									(and
+										;not a time-series feature (lag, rate, etc.)
+										(not (contains_index (get !featureAttributes (current_value)) "ts_type"))
+										;generated upon training, otherwise not saved into cases
+										(size (get !featureAttributes [(current_value 1) "auto_derive_on_train"]) )
+										;at least one source feature that affects this derived feature was edited
+										(size (filter
+											(lambda
+												;"list of derived features contains this specific derived feature?""
+												(contains_value (current_value) (current_value 1))
+											)
+											;only iterate over sources features that were edited
+											(keep !sourceToDerivedFeatureMap features)
+										))
+										;the derivation code only uses values from within the same case (not other cases in the series)
+										;(get_all_labels) returns a map of all labels to their values from the code, in this case all feature names to their  offsets,
+										;offsets other than zero indicate deriving from values of another row within a series
+										;tackling this problem for TS data is TODO: 21518
+										(=
+											[0]
+											(values
+												(get_all_labels (parse (get !featureAttributes (list (current_value 1) "auto_derive_on_train" "code"))))
+												(true)
+											)
+										)
+									)
+								)
+								(indices !derivedFeaturesMap)
```

Comment:
you could (or) this with an [ ] to ensure that affected_derived_features is always a list and never a null.

Reply:
sure